### PR TITLE
refactor: extract crypto/fiat selector/amount components

### PR DESCRIPTION
**Pull request by mrfelton**
## Description:

- Create new containers for `CrytoSelector`, `CryptoValue`, `FiatSelector`, `FiatValue` and update throughout the app.

- Update storybook to use redux state (required since components that we render in the storybook now depend on redux)

## Motivation and Context:

We use these crypto/fiat dropdowns and display crypto/fiat amounts in many places. Currently we need to build them and pass props to them manually every time. This creates a lot of duplicate code and repetitive prop passing.

## How Has This Been Tested?

Ensure all tests pass
Test app manually
Ensure storybook works

## Types of changes:
Refactor

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have reviewed and updated the documentation accordingly.
- [x] I have read the _CONTRIBUTING_ document.
- [ ] I have added tests to cover my changes where needed.
- [x] All new and existing tests passed.
- [x] My commits have been squashed into a concise set of changes.

### CODE REVIEW

File: app/components/Activity/TransactionModal/TransactionModal.js [hunk 1]
```diff
@@ -176,7 +117,9 @@ export default class TransactionModal extends React.PureComponent {
                 <Link
                   className="hint--bottom-left"
                   data-hint={item.block_hash}
-                  onClick={() => blockExplorer.showBlock(network, item.block_hash)}
+                  onClick={() =>
+                    networkInfo && blockExplorer.showBlock(networkInfo, item.block_hash)
```
Comment (korhaliv) [inline app/components/Activity/TransactionModal/TransactionModal.js]: This pattern is reused across this file. May be a good idea to extract it like these for the ease of change and cleanness (untested)
```
const showBlock = (hash) => {
  return networkInfo && blockExplorer.showTransaction(networkInfo, hash)	
} 

```

File: stories/containers/pay.stories.js [hunk 2]
```diff
@@ -55,100 +35,100 @@ const store = new Store({
...
-          total_fees_msat: '3',
-          total_amt_msat: '10000021'
-        }
-      ]
-    })
-  } else {
-    store.set({ routes: [] })
-  }
-  store.set({ isQueryingRoutes: false })
+  // data.set({ isQueryingRoutes: true })
```
Comment (korhaliv) [inline stories/containers/pay.stories.js]: Is this commented code is intended to stay or was forgotten to be removed?

File: app/components/Activity/ActivityModal/ActivityModal.js [hunk 3]
```diff
@@ -2,35 +2,49 @@ import React from 'react'
...
+        component: PaymentModal,
+        props: { item }
+      },
+      invoice: {
+        component: InvoiceModal,
+        props: { item, showNotification }
+      }
+    }
 
     if (!item) {
```
[the remark this answers is not in the document]
Comment (mrfelton) [reply app/components/Activity/ActivityModal/ActivityModal.js]: Good call. Updated

File: app/components/Activity/TransactionModal/TransactionModal.js (hunk 1 above)
Comment (mrfelton) [reply app/components/Activity/TransactionModal/TransactionModal.js]: Updated.

File: stories/containers/pay.stories.js (hunk 2 above)
Comment (mrfelton) [reply stories/containers/pay.stories.js]: Meant to be removed - have updated

File: app/components/Activity/Payment/Payment.js [hunk 4]
```diff
@@ -1,20 +1,12 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 import { Box, Flex } from 'rebass'
-import { btc } from 'lib/utils'
-import { Message, Span, Text, Value } from 'components/UI'
-import { FormattedMessage, FormattedNumber, FormattedTime, injectIntl } from 'react-intl'
+import { Message, Span, Text } from 'components/UI'
+import { CryptoValue, FiatValue } from 'containers/UI'
+import { FormattedMessage, FormattedTime, injectIntl } from 'react-intl'
```
[the remark this answers is not in the document]
Comment (mrfelton) [reply app/components/Activity/Payment/Payment.js]: Updated

**Comment by mrfelton** [comment]
Thanks @korhaliv . I've made those edits.

**Pull request merged**